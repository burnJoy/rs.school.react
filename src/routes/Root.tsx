import React from 'react';
import { Outlet } from 'react-router-dom';

class Root extends React.Component {
  setPageTitle(title: string) {
    console.log(title);
  }
  render() {
    return (
      <>
        <h1>page title</h1>
        <div className="content">
          <Outlet context={[this.setPageTitle]} />
        </div>
      </>
    );
  }
}
export default Root;
