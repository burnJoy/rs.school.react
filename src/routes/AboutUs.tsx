import React from 'react';
import { TitleContext, type Props } from './Root';

class AboutUs extends React.Component<Props> {
  static contextType = TitleContext;
  declare context: React.ContextType<typeof TitleContext>;

  componentDidMount() {
    this.context!(this.props.title);
  }

  render() {
    return (
      <>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Sequi quasi, recusandae ipsa
          deserunt exercitationem aliquid consequatur esse blanditiis beatae neque quia dolore quis
          facilis voluptatum. Animi exercitationem temporibus eaque laudantium minima doloribus
          atque modi nihil numquam, aliquam laborum a assumenda repudiandae officia est fuga quia
          dignissimos facere delectus unde voluptate deserunt? Recusandae aperiam debitis impedit
          reprehenderit ad tempora eaque odit minus, distinctio fugit! Exercitationem perferendis
          doloribus, natus tempore quos unde provident enim debitis ullam harum, quae itaque
          repellendus, illum ducimus soluta nihil corporis reprehenderit recusandae iusto. Assumenda
          quisquam iste, sequi necessitatibus amet deserunt veritatis qui voluptate ipsum,
          consectetur, harum quaerat!
        </p>
      </>
    );
  }
}

export default AboutUs;
