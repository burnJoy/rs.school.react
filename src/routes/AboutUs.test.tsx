import React from 'react';
import { vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { TitleContext } from './Root';
import AboutUs from './AboutUs';

describe('About us page', () => {
  const title = 'test title for about us page';

  it('provider function is called', () => {
    const mockFunc = vi.fn();
    render(
      <TitleContext.Provider value={mockFunc}>
        <AboutUs title={title} />
      </TitleContext.Provider>
    );

    expect(mockFunc).toBeCalledTimes(1);
    expect(mockFunc).toBeCalledWith(title);
  });

  it('matches the snapshot', () => {
    const { container } = render(
      <TitleContext.Provider value={() => {}}>
        <AboutUs title={title} />
      </TitleContext.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
