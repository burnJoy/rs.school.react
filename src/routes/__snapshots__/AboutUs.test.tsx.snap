// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About us page > matches the snapshot 1`] = `
<p>
  Lorem ipsum dolor sit amet consectetur adipisicing elit. Sequi quasi, recusandae ipsa deserunt exercitationem aliquid consequatur esse blanditiis beatae neque quia dolore quis facilis voluptatum. Animi exercitationem temporibus eaque laudantium minima doloribus atque modi nihil numquam, aliquam laborum a assumenda repudiandae officia est fuga quia dignissimos facere delectus unde voluptate deserunt? Recusandae aperiam debitis impedit reprehenderit ad tempora eaque odit minus, distinctio fugit! Exercitationem perferendis doloribus, natus tempore quos unde provident enim debitis ullam harum, quae itaque repellendus, illum ducimus soluta nihil corporis reprehenderit recusandae iusto. Assumenda quisquam iste, sequi necessitatibus amet deserunt veritatis qui voluptate ipsum, consectetur, harum quaerat!
</p>
`;
